kite-data/kite-data-core/src/main/java/org/kitesdk/data/View.java

```diff
@@ -109,6 +111,15 @@
   public Class<E> getType();
 
   /**
+   * Get the schema of entities contained in this {@code View}.
+   *
+   * @return the schema of entities contained in this view
+   *
+   * @since 1.1.0
+   */
+  public Schema getSchema();
```

Comment:
Okay, thinking about this more... I don't think my suggestion is valid.

The schema you set on the view is the view's schema and isn't necessarily the a read schema. The dataset's schema must always be able to read that schema, but it can be more strict that the dataset's schema. It can also be more permissive, in the case where you're projecting a subset of the columns you don't care about the ones that aren't being read.

So then we can't validate the schema from `asSchema` when it is used to create a view, only when we use that view to get a reader or a writer. (Which we should do, by the way.) Because the schema could be either a read or write schema, we can't name this method for either one. But we still need to distinguish it from the dataset's schema. `getViewSchema` or `getEffectiveSchema`?